safe.go

```diff
@@ -0,0 +1,72 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package waf
+
+import (
+	"fmt"
+	"reflect"
+	"runtime"
+
+	"github.com/pkg/errors"
+)
+
+// PanicError is an error type wrapping a recovered panic value that happened
+// during a function call. Such error must be considered unrecoverable and be
+// used to try to gracefully abort. Keeping using this package after such an
+// error is unreliable and the caller must rather stop using the library.
+// Examples include safety checks errors.
+type PanicError struct {
+	// The function that was given to `Call()`.
+	In func() error
+	// The recovered panic value while executing `In()`.
+	Err error
```

Comment:
Is there a need for those to be public? Seems like we could "leak" some internal functions through this which seems like an odd behaviour.
Also, what about having newPanicError embed the function's name instead, since it seems to be the purpose of embedding the function here? It's less "lazy" but we don't really need to be lazy for this since a panic occurred.

Reply:
Good points yes, it was done in 82c1a4d32b97f515876efc61490831b7d76120bb
I kept the `Err` field public as this is a wrapper error type where the usual "paradigm" is to give access to the underlying wrapped error.